packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -63,3 +99,39 @@ function relativeOffset(ancestor: HTMLElement, child: HTMLElement, axis: 'left'|
   }
   return sum;
 }
+
+interface ScrollIntoViewportOpts {
+  /** The optional containing element of the target to be centered in the viewport. */
+  containingElement?: Element
+}
+
+/**
+ * Scrolls the `targetElement` so it is visible in the viewport. Accepts an optional `opts.containingElement`
+ * that will be centered in the viewport prior to scrolling the targetElement into view. If scrolling is prevented on
+ * the body (e.g. targetElement is in a popover), this will only scroll the scroll parents of the targetElement up to but not including the body itself.
+ */
+export function scrollIntoViewport(targetElement: Element, opts?: ScrollIntoViewportOpts) {
+  // If scrolling is not currently prevented then we aren’t in a overlay nor is a overlay open, just use element.scrollIntoView to bring the element into view
+  if (!isScrollPrevented) {
+    let {left: originalLeft, top: originalTop} = targetElement.getBoundingClientRect();
+
+    // use scrollIntoView({block: 'nearest'}) instead of .focus to check if the element is fully in view or not since .focus()
+    // won't cause a scroll if the element is already focused and doesn't behave consistently when an element is partially out of view horizontally vs vertically
+    targetElement?.scrollIntoView?.({block: 'nearest'});
+    let {left: newLeft, top: newTop} = targetElement.getBoundingClientRect();
+    // Account for sub pixel differences from rounding
+    if ((Math.abs(originalLeft - newLeft) > 1) || (Math.abs(originalTop - newTop) > 1)) {
+      opts?.containingElement?.scrollIntoView?.({block: 'center', inline: 'center'});
+      targetElement.scrollIntoView?.({block: 'nearest'});
+    }
+  } else {
+    let root = document.scrollingElement || document.documentElement;
+    let scrollParent = getScrollParent(targetElement);
+    // If scrolling is prevented, we don't want to scroll the body since it might move the overlay partially offscreen and the user can't scroll it back into view.
```

Comment:
What's the case where this is needed? Don't we set `overflow: hidden` on the body, so it wouldn't be scrollable anyway? What happens if you just call `element.scrollIntoView`?

Reply:
Even if `overflow: hidden` is set on the body, `element.scrollIntoView` will can still scroll the element into view. 

https://user-images.githubusercontent.com/9661127/213599990-a81f6c77-6cbb-42f0-8a17-5a98e2630ab9.mov

The current case where this is needed is if the Datepicker popover is opened off screen, causing the entire page to scroll to bring it into view if we just use `element.scrollIntoView`, something that maybe I can mitigate using a long timeout or something. However, this may become a problem down the line if there ever is a case where usePreventScroll is active and the user performs some interaction that calls scrollIntoViewport